pygmt/tests/test_grdfill.py

```diff
@@ -0,0 +1,44 @@
+"""
+Tests for grdclip.
+"""
+import os
+
+import numpy as np
+import pytest
+import xarray as xr
+from pygmt import grdfill, grdinfo
+from pygmt.datasets import load_earth_relief
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file and set value(s) to
+    NaN.
+    """
+    grid = load_earth_relief(registration="pixel")
+    grid[10, 10] = np.nan
+    return grid
+
+
+def test_grdfill_dataarray_out(grid):
+    """
+    grdfill with a DataArray output.
+    """
+    result = grdfill(grid=grid, mode="c20")
+    # check information of the output grid
+    assert isinstance(result, xr.DataArray)
+    assert result[10, 10] == 20
```

Comment:
Add some extra checks to see that the whole grid does not have NaN values.

```suggestion
    assert result[4, 4] == 20
    assert result[5, 5] == np.inf
    assert not result.isnull().all()  # check that no NaN values exists
    assert result.gmt.gtype == 1  # Geographic grid
    assert result.gmt.registration == 1  # Pixel registration
```